exporter/collector/monitoredresource.go

```diff
@@ -71,7 +75,15 @@ func resourceToLabels(
 		}
 		// Matches one of the resource filters
 		for _, resourceFilter := range resourceFilters {
-			if strings.HasPrefix(k, resourceFilter.Prefix) {
+			matchesRegex := true
+			if len(resourceFilter.Regex) > 0 {
+				matched, err := regexp.Match(resourceFilter.Regex, []byte(k))
+				if err != nil {
+					log.Debug("Unable to parse resource filter regex.", zap.Error(err), zap.Any("regex", resourceFilter.Regex))
```

Comment:
should parse errors be found on startup (and be fatal), instead of logged each time we try to match something?